client/clientutil/modelinfo.go

```diff
@@ -0,0 +1,455 @@
+			}
+
+		// timestamp needs to be formatted in an identical manner to how fmtTime works
+		// from timeMixin package in cmd/snap
+		case "timestamp":
+			timestamp, ok := headerValue.(string)
+			if !ok {
+				return formatInvalidTypeErr(headerName)
+			}
+
+			// parse the time string as RFC3339, which is what the format is
+			// always in for assertions
+			t, err := time.Parse(time.RFC3339, timestamp)
+			if err != nil {
+				return err
+			}
+			fmt.Fprintf(w, "timestamp:\t%s\n", fmtTime(t, options.AbsTime))
+
+		// long string key we don't want to rewrap but can safely handle
+		// on "reasonable" width terminals
+		case "device-key-sha3-384":
+			// also flush the writer before continuing so the previous keys
+			// don't try to align with this key
+			w.Flush()
+			headerString, ok := headerValue.(string)
+			if !ok {
+				return formatInvalidTypeErr(headerName)
+			}
+
+			switch {
+			case options.TermWidth > 86:
+				fmt.Fprintf(w, "device-key-sha3-384: %s\n", headerString)
+			case options.TermWidth > 66:
+				fmt.Fprintln(w, "device-key-sha3-384: |")
+				strutil.WordWrapPadded(w, []rune(headerString), "  ", options.TermWidth)
+			}
+		case "snaps":
+			// also flush the writer before continuing so the previous keys
+			// don't try to align with this key
+			w.Flush()
+			snapsHeader, ok := headerValue.([]interface{})
+			if !ok {
+				return formatInvalidTypeErr(headerName)
+			}
+			if len(snapsHeader) == 0 {
+				// unexpected why this is an empty list, but just ignore for
+				// now
+				continue
+			}
+
+			fmt.Fprintf(w, "snaps:\n")
+			if err := printVerboseSnapsList(w, snapsHeader); err != nil {
+				return err
+			}
+
+		// long base64 key we can rewrap safely
+		case "device-key":
+			headerString, ok := headerValue.(string)
+			if !ok {
+				return formatInvalidTypeErr(headerName)
+			}
+			// the string value here has newlines inserted as part of the
+			// raw assertion, but base64 doesn't care about whitespace, so
+			// it's safe to replace the newlines
+			headerString = strings.ReplaceAll(headerString, "\n", "")
+			fmt.Fprintln(w, "device-key: |")
+			strutil.WordWrapPadded(w, []rune(headerString), "  ", options.TermWidth)
+
+		// The rest of the values should be single strings
+		default:
+			headerString, ok := headerValue.(string)
+			if !ok {
+				return formatInvalidTypeErr(headerName)
+			}
+			fmt.Fprintf(w, "%s:\t%s\n", headerName, headerString)
+		}
+	}
+	return w.Flush()
+}
+
+// PrintModelAssertionYAML will format the provided serial or model assertion based on the parameters given in
+// YAML format. The output will be written to the provided io.Writer.
+func PrintModelAssertionYAML(w *tabwriter.Writer, modelFormatter ModelFormatter, options PrintModelAssertionOptions, modelAssertion asserts.Model, serialAssertion *asserts.Serial) error {
+
+	// if assertion was requested we want it raw
+	if options.Assertion {
+		_, err := w.Write(asserts.Encode(&modelAssertion))
```

Comment:
So not YAML after all, Perhaps this should be handled separately? eg `func PrintAssertionRaw(w io.Writer, a *asserts.Assertion)` so that you can use it for serial too.

Reply:
I would agree with this yes, but it will create inconsistensies in handling of the Assertion option which would make the code more confusing in my opinion, but I will happily change it if you feel strongly about this.

Reply:
Updated, and removed the 'YAML'